torchvision/prototype/transforms/_auto_augment.py

```diff
@@ -43,29 +32,31 @@ def _get_random_item(self, dct: Dict[K, V]) -> Tuple[K, V]:
         key = keys[int(torch.randint(len(keys), ()))]
         return key, dct[key]
 
-    def _extract_image(
+    def _flatten(
```

Comment:
I understand that this is an internal method but I feel its name and return types can lead to confusion to someone reading the code. This method now ends up doing too many things (flattening and extract image); similar for unflatten. What is the reason for this?

Reply:
> This method now ends up doing too many things (flattening and extract image)

Basically it does the same thing as before. As explained in https://github.com/pytorch/vision/pull/6434#discussion_r948738057 `idx_flat` corresponds to the flattened sample. Thus, if we don't returned the flattened sample, we need to flatten it again later when we put the augmented image in. 

> I understand that this is an internal method but I feel its name and return types can lead to confusion to someone reading the code.

Agreed. I'll come up with better names.

Reply:
So this is more of a speed optimization? I don't know if flattening and unflattening can be a bottleneck. If we believe it is, we should measure it.

My concern here is that the previous code, though it was performing extra ops, it was much clearer. The new one, due to its inconsistent naming and complexity, is much harder to read.